### PR TITLE
Fix for timeout not set when not calling Init()

### PR DESCRIPTION
**Pull request by ldemailly**

### CODE REVIEW

**Comment by ldemailly** [comment]
@mandarjog this will fix the timeout in rate limit
I didn't review #65 carefully enough thinking about earlier code not calling Init()

File: fhttp/http_test.go [hunk 1]
```diff
@@ -80,6 +80,25 @@ func TestNewHTTPRequest(t *testing.T) {
 	}
 }
 
+func TestMultiInitAndEscape(t *testing.T) {
+	// one escaped already 2 not
+	o := NewHTTPOptions("http://localhost:8080/?delay=1s:10%,0.5s:15%25,0.25s:5%")
```
Comment (rkpagadala) [inline fhttp/http_test.go:85]: suggestion:
expected := "http://localhost:8080/?delay=1s:10%,0.5s:15%25,0.25s:5%"
o := NewHTTPOptions(expected)

File: fhttp/http.go [hunk 2]
```diff
@@ -216,6 +221,10 @@ func NewStdClient(o *HTTPOptions) Fetcher {
 	if o.NumConnections < 1 {
 		o.NumConnections = 1
 	}
+	// 0 timeout for stdclient doesn't mean 0 timeout... so just warn and leave it
```
Comment (rkpagadala) [inline fhttp/http.go:224]: explain what the effects of timeout of 0, negative value

**Review by rkpagadala — APPROVED** [review]

File: fhttp/http_test.go (hunk 1 above)
Comment (ldemailly) [reply fhttp/http_test.go:85]: expected is not same as input

Comment (rkpagadala) [reply fhttp/http_test.go:85]: ok. it was hard to tell the difference.

Comment (ldemailly) [reply fhttp/http_test.go:85]: yeah it’s a bit tricky those %
(sorry, terse as i have been on the phone)

File: fhttp/http.go (hunk 2 above)
Comment (ldemailly) [reply fhttp/http.go:224]: tbh I have to look it up
initially my test didn’t fail when I commented out the code that was setting the timeout to the default like I did for the fast custom client 

and tried with -1 and that also passed locally, I assumed it meant infinite timeout but then it failed on circle, so I set it back to unitialized/0

**Pull request merged**